consensus/src/units/store.rs

```diff
@@ -99,12 +87,12 @@ impl<'a, H: Hasher, D: Data, KB: KeyBox> UnitStore<'a, H, D, KB> {
         let top_row = NodeMap::with_hashmap(n_nodes, gm.clone().max());
         let first_missing_rounds = NodeMap::with_hashmap(
             n_nodes,
-            gm.collect::<Vec<_>>()
+            gm.collect::<HashSet<_>>()
                 .into_iter()
-                .map(|(id, rounds)| (id, first_missing(rounds)))
-                .filter(|(id, round)| match top_row.get(*id) {
-                    Some(top_round) => round < top_round,
-                    None => false,
+                .filter_map(|(id, rounds)| {
+                    (0..*top_row.get(id).unwrap())
```

Comment:
Oh no, not `unwrap`! D:

Wrap this in a `match top_row.get(id)` and `None => None,`? Not sure what clippy will have to say about that tho. <_<

Reply:
`unwrap_or(0)` would also be acceptable, although annoying implicit.

Reply:
matched